internals/overlord/state/notices.go

```diff
@@ -279,6 +307,9 @@ func (f *NoticeFilter) matches(n *Notice) bool {
 	if f == nil {
 		return true
 	}
+	if f.UserID != nil && n.userID != nil && *f.UserID != *n.userID {
```

Comment:
This logic seems sound, but it took me a couple of takes to confirm that. Perhaps we can add a comment to help the reader. Something like:

```suggestion
	// Three cases here:
	// - if there is no user ID filter, include the notice (whether it's public or private)
	// - if the notice's user ID is not set, the notice is public, so include it
	// - if there is a user ID filter and the notice user ID is set, only include it if they match
	if f.UserID != nil && n.userID != nil && *f.UserID != *n.userID {
```

Reply:
I considered the following as well:
```go
if f.UserID != nil && !(n.userID == nil || *n.userID == *f.UserID) {
```

I'll include the comment for now, and if you prefer this code suggestion, I can add it as well.

Reply:
Yeah, actually, I prefer that "filter exists and !matches" logic -- need for the comment almost goes away in that case I think.

Reply:
Sounds good. I've made the switch and removed the comment --- I think having both is a bit confusing, since the comment describes the complement (?) of the actual check.